src/ffi/dial_ffi.rs

```diff
@@ -172,18 +175,19 @@ pub unsafe extern "C" fn dial(
     } else {
         disable_webrtc = uri_str.contains(".local") || uri_str.contains("localhost");
     }
-    let server = match runtime.block_on(async move {
+    let (server, channel) = match runtime.block_on(async move {
         let dial = match payload {
-            Some(p) => tower::util::Either::A(
+            Some(p) => Either::A(
                 dial_with_cred(uri_str, p.to_str()?, allow_insec, disable_webrtc)?
                     .connect()
                     .await?,
             ),
             None => {
                 let c = dial_without_cred(uri_str, allow_insec, disable_webrtc)?;
-                tower::util::Either::B(c.connect().await?)
+                Either::B(c.connect().await?)
             }
         };
+        let channel = dial.clone();
```

Comment:
`channel` is defined here but not used again until line 207. Mild preference for defining it around line 206, assuming that doesn't break functionality (and if it would, please document why it needs to be way up here). 

```rust
...
let server = ...
let channel = dial.clone();
OK(server, channel)
```

Reply:
channel can't be defined lower because the immediate following line (`let g = GRPCProxy::new(dial, uri);`) takes ownership of the dial.

Reply:
actually wait I think we can make this cleaner. just pushed, lemme know what you think

Reply:
Yes, I like this more! Thanks